builtin/init-db.c

```diff
@@ -258,15 +258,24 @@ static int create_default_files(const char *template_path,
 		die("failed to set up refs db: %s", err.buf);
 
 	/*
-	 * Create the default symlink from ".git/HEAD" to the "master"
-	 * branch, if it does not exist yet.
+	 * Create the default symlink from ".git/HEAD" to the default
+	 * branch name, if it does not exist yet.
 	 */
 	path = git_path_buf(&buf, "HEAD");
 	reinit = (!access(path, R_OK)
 		  || readlink(path, junk, sizeof(junk)-1) != -1);
 	if (!reinit) {
-		if (create_symref("HEAD", "refs/heads/master", NULL) < 0)
+		char *default_branch_name;
+		git_config_get_default_branch_name(&default_branch_name);
```

Comment:
I don't quite follow what you're suggesting here (the return statement is confusing me…). Are you suggesting replacing the method `git_config_get_default_branch_name()` with the sample code? In `git_config_get_default_branch_name()`, I am checking the env, but only after checking the config, which seems to be the pattern for config vars. It appears you are also suggesting reversing the priority here?

Reply:
Personally, I like the helper method to avoid extra complexity here.

The fact that `git_config_get_string_const()` returns a negative value means something went horribly wrong while parsing config, so we should probably halt here. However, it would be better to signal an error higher up.

In this case, I don't think the extra care here is merited. Perhaps we should just do the following?

```
const char *branch_name;

if ((branch_name = getenv("GIT_TEST_DEFAULT_BRANCH_NAME") &&
    *branch_name)
        return branch_name;

if (git_config_get_string_const("init.defaultbranchname", &branch_name))
        return branch_name;

return "master";
```

(nit: replace my spaces with tabs)

Reply:
That's a really good suggestion.